src/huggle_ui/blockuserform.cpp

```diff
@@ -133,6 +133,7 @@ void BlockUserForm::Block()
     this->timer->stop();
     if (this->ui->cbMessageTarget->isChecked())
         this->sendBlockNotice(nullptr);
```

Comment:
Here if user wants to send a message to user who was blocked after blocking them, function sendBlockNotice is called, you can see from "this" that this function is a member function of this very class. However on next line you close this window and since it has "deleteOnClose" attribute set, the class will be deleted, before ever finishing (and verifying) the notice sending operation.

Easiest fix would be to put else statement there and close window only when we don't want to message the user.

Reply:
ok I take it back `sendBlockNotice` doesn't set any callback to verify the results, so it's safe to delete the class after calling it